### PR TITLE
moving Job/Task framework out of 'render' library and into its own 'task' library

### PR DESCRIPTION
**Pull request by AndrewMeadows**
Also, creating a 'workload' library that will rely on 'task' and will categorize objects by proximity so we can adjust load levels.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7026/app_12365_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7026/HighFidelity-Beta-PR12365-691cd017c662f73bcba38524c5ac3fc053c7cb6d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7026/HighFidelity-Beta-PR12365-691cd017c662f73bcba38524c5ac3fc053c7cb6d.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7037/app_12365_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7037/HighFidelity-Beta-PR12365-7a72174eb7f59b562d5834c36b51f0097f9dda49.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7058/app_12365_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7058/HighFidelity-Beta-PR12365-7a72174eb7f59b562d5834c36b51f0097f9dda49.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7079/app_12365_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7080/app_12365_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7098/app_12365_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7098/HighFidelity-Beta-PR12365-0e38d7ee635a97ec2873f5f6f4ef302a5f528917.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7105/app_12365_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7106/app_12365_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7106/HighFidelity-Beta-PR12365-0d26c67417dc5bf6ebcee8d695281d4a121cc089.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7106/HighFidelity-Beta-PR12365-0d26c67417dc5bf6ebcee8d695281d4a121cc089.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7117/app_12365_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7117/HighFidelity-Beta-PR12365-9b33ea7deaebeb06d98bcc97385b05036ab5bbc6.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7117/HighFidelity-Beta-PR12365-9b33ea7deaebeb06d98bcc97385b05036ab5bbc6.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7124/app_12365_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7124/HighFidelity-Beta-PR12365-4ae829c0acb847c123bad80b13673400923dce73.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7124/HighFidelity-Beta-PR12365-4ae829c0acb847c123bad80b13673400923dce73.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7126/app_12365_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7126/HighFidelity-Beta-PR12365-12502bce521cde1fb1a8554a72662999d0ae8f40.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7126/HighFidelity-Beta-PR12365-12502bce521cde1fb1a8554a72662999d0ae8f40.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7133/app_12365_PR.apk

**Comment by AndrewMeadows** [comment]
What has changed:

(1) The **workload** library has been added but isn't actually used anywhere yet.  The interface-client depends on it for build/link, but doesn't yet instantiate anything from it.
(2) The **task** library has been lifted out of **render** library and is otherwise unchanged except for how it gets the `PROFILE` context for profile debugging, which is a dev-debug-only thing right now.

**Test plan:**

If the interface-client still renders stuff normally then everything is good.  The **render** pipeline depends on the **task** stuff to run the various stages of sorting, culling, sending of geometry to the graphics card:

Make sure you can login and still: look around and see content rendered as expected.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7133/HighFidelity-Beta-PR12365-3af19b730639c51d4c3b7875299119ecd8ecbfde.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7133/HighFidelity-Beta-PR12365-3af19b730639c51d4c3b7875299119ecd8ecbfde.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7154/app_12365_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7154/HighFidelity-Beta-PR12365-01fb8f336956527fbe68a29f8c71a44d1e557c1b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7154/HighFidelity-Beta-PR12365-01fb8f336956527fbe68a29f8c71a44d1e557c1b.exe

File: libraries/entities/src/EntityTree.cpp [hunk 1]
```diff
@@ -913,18 +913,25 @@ void EntityTree::findEntities(RecurseOctreeOperation& elementFilter,
...
+    }
+    if (foundEntity && !foundEntity->getElement()) {
+        // special case to maintain legacy behavior:
+        // if the entity is in the map but not in the tree
+        // then pretend the entity doesn't exist
+        return EntityItemPointer(nullptr);
+    } else {
+        return foundEntity;
     }
-    return foundEntity;
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityTree.cpp]: While looking at other stuff I noticed this circuitous logic in our code: when looking for an `EntityItem` by ID we would (1) find the item in the `_entityMap`, (2) get its `EntityTreeElement`, (3) walk entities in `EntityTreeElement` to find the entity again.

The new code will now: (1) find the `EntityItem` in the the `_entityMap` (2) return the entity if it has been added to any `EntityTreeElement`, else return `nullptr`.

The behavior hasn't changed but the path to get there is shorter.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7161/HighFidelity-Beta-PR12365-20c2a4470e8c2bba6c3b72893428a2ea1ec89e6f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7161/HighFidelity-Beta-PR12365-20c2a4470e8c2bba6c3b72893428a2ea1ec89e6f.exe

**Comment by AndrewMeadows** [comment]
I did the smoke testing on this: Windows-desk, Windows-HMD, Linux-desk.  It is good.

**Pull request merged**